hail/python/hail/backend/service_backend.py

```diff
@@ -325,6 +325,49 @@ def render(self, ir):
         assert len(r.jirs) == 0
         return r(finalize_randomness(ir))
 
+    async def _submit_batch(self, build_batch_f, name, *, progress: Optional[BatchProgressBar] = None, **batch_args):
```

Comment:
Hmm. I don't think this belongs here, it only references a handful of properties from ServiceBackend. It's totally legitimate for `_service_vep` to ask for things like `hl.current_backend().name_prefix` etc.

If we want to use some of the QoB error formatting information elsewhere, perhaps that should live in `hailtop.batch`?

Reply:
I think my inclination is to just stuff this in service_vep until we have another example of Batch-QoB interaction. A few more examples will enable us to generalize appropriately.